src/libusermode/userhook.cpp

```diff
@@ -855,6 +856,20 @@ bool drakvuf_request_usermode_hook(drakvuf_t drakvuf, const dll_view_t* dll, con
     return true;
 }
 
+std::optional<LogStrategy> get_log_strategy(const std::string& str)
```

Comment:
I don't think LogStrategy is a good name. My understanding of `dll_hooks` is that sometimes plugins authors want to leave some space for the end-users to define which calls should trigger which actions (example `log`, `stack`). And so in `apimon` we want the user to be able to customize the list of apicalls which should be logged and in `memdump` we want the user to be able to customize the list of apicalls which will perform the `stack walk`.

Reply:
I'm open to suggestions and personally not really satisfied with the name, but I wasn't able to come up with anything better ;)

Reply:
Maybe "trace"?

Reply:
After a discussion with @kscieslinski, I've decided to make it a `struct` to be more easily extendable in the future.